tests/click_types.py

```diff
@@ -0,0 +1,163 @@
+"""
+check annotations on click commands
+requires python3.8+
+"""
+from __future__ import annotations
+
+import datetime
+import types
+import typing as t
+import uuid
+
+import click
+
+from globus_cli.constants import ExplicitNullType
+from globus_cli.parsing.known_callbacks import none_to_empty_dict
+from globus_cli.parsing.param_types import (
+    CommaDelimitedList,
+    EndpointPlusPath,
+    IdentityType,
+    JSONStringOrFile,
+    LocationType,
+    NotificationParamType,
+    ParsedIdentity,
+    StringOrNull,
+    TaskPath,
+    TimedeltaType,
+    UrlOrNull,
+)
+from globus_cli.types import JsonValue
+
+
+def _param2types(param_obj: click.Parameter) -> tuple[type, ...]:
+    param_type = param_obj.type
+
+    # click types
+    if isinstance(param_type, click.types.StringParamType):
+        return (str,)
+    if isinstance(param_type, click.types.BoolParamType):
+        return (bool,)
+    elif isinstance(param_type, (click.types.IntParamType, click.IntRange)):
+        return (int,)
+    elif isinstance(param_type, (click.types.FloatParamType, click.FloatRange)):
+        return (float,)
+    elif isinstance(param_type, click.Choice):
+        return (t.Literal[tuple(param_type.choices)],)
```

Comment:
I'm not familiar with this syntax. Does this tuple unpacking, or is it valid as-is?

```suggestion
        return (t.Literal[*tuple(param_type.choices)],)
```

Reply:
Apparently, both of these are valid. The argument to `__class_getitem__` is a tuple, but it looks like unpacking is automatic. Here's a little sample from the console:
```
>>> import typing as t
>>> x = (int, str)
>>> y = t.Literal[x]
>>> y
typing.Literal[int, str]
>>> z = t.Literal[*x]
>>> z
typing.Literal[int, str]
>>> w = t.Literal[(x,)]
>>> w
typing.Literal[(<class 'int'>, <class 'str'>)]
```

So that looks like `y` and `z` are equivalent, but `w` is improperly formed/invalid.

I don't have a preference between passing a tuple or an unpacked tuple, but I'd like to be consistent in how we do these constructions. Given the choice, which one would you prefer?

---

I find the syntax here a little odd, and I think the use-cases (like this one) for dynamically constructing types somewhat surprised the original authors of the type annotation syntax and PEPs. However, given the recent course-corrections the SC has made with respect to typing and wanting to support runtime usages, I feel okay about using this.